bootstrap/kubeadm/api/v1beta1/kubeadmconfig_types.go

```diff
@@ -107,6 +121,219 @@ type KubeadmConfigSpec struct {
 	Ignition *IgnitionSpec `json:"ignition,omitempty"`
 }
 
+// Validate ensures the KubeadmConfigSpec is valid.
+func (c *KubeadmConfigSpec) Validate(pathPrefix *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+
+	allErrs = append(allErrs, validateFiles(*c, pathPrefix)...)
+	allErrs = append(allErrs, validateUsers(*c, pathPrefix)...)
+	allErrs = append(allErrs, validateIgnition(*c, pathPrefix)...)
+
+	return allErrs
+}
+
+func validateFiles(c KubeadmConfigSpec, pathPrefix *field.Path) field.ErrorList {
```

Comment:
```suggestion
func (c *KubeadmConfigSpec) validateFiles(pathPrefix *field.Path) field.ErrorList {
```

I would suggest to keep these 3 funcs as methods. I don't see the benefit in passing in KubeadmConfigSpec

(maybe I'm missing something)

Reply:
Yes, makes sense. I will change that.